compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1750,6 +1751,40 @@ impl VariantDef {
     }
 }
 
+impl PartialOrd for VariantDef {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(&other))
+    }
+}
+
+/// There should be only one VariantDef for each `def_id`, therefore
+/// it is fine to implement `Ord` only based on `def_id`.
+impl Ord for VariantDef {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.def_id.cmp(&other.def_id)
```

Comment:
Please do not compare `DefId`.  This could break incremental compilation.  We have been working on removing the `impl PartialOrd for DefId`.

Reply:
Thanks for letting me know!

These impls are here because I need to store `VariantDef`s in a set (either `HashSet` or `BTreeSet`). Can you recommend an alternative way of doing this?

Reply:
Using a FxHashSet (or FxIndexSet if you need to iterate on it) is fine.